compiler/src/lntoamm/UserFunction.ts

```diff
@@ -642,9 +642,11 @@ ${statements[i].statementAst.t.trim()} on line ${statements[i].statementAst.line
 
   microstatementInlining(
     realArgNames: Array<string>,
-    scope: Scope,
+    s: Scope,
     microstatements: Array<Microstatement>,
   ) {
+    const scope = new Scope(s)
+    scope.secondaryPar = this.scope
```

Comment:
I changed it so function inlining and dispatching get access to both the current scope and the scope they were originally created in to make sure unexported types are available for use. Hopefully this doesn't break something else, this is the piece I'm most worried about.